src/test/java/com/aws/greengrass/clientdevices/auth/helpers/TestHelpers.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.helpers;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+/**
+ * Helpers for generic assertion tooling
+ */
+public final class TestHelpers {
+    public static final long DEFAULT_GENERIC_POLLING_TIMEOUT_MILLIS =
+            TimeUnit.SECONDS.toMillis(10);
+
+    private TestHelpers() {
+    }
+
+    public static boolean eventuallyTrue(Supplier<Boolean> condition, long... optional) throws InterruptedException {
```

Comment:
Caller of this method won't know what these long optional params correspond to, unless they check its code. I see the first long value corresponds to the timeout in millis, second long value is polling interval in millis. I'd suggest having these as separate parameter values than optionals (`long... optional`).

We can expose two overloaded methods:
```
public eventuallyTrue(Supplier<Boolean> condition, long timeoutMillis, long pollingIntervalMillis);
public eventuallyTrue(Supplier<Boolean> condition);  // uses default values for timeout and pollingInterval
```